src/Control/Alternative/Free.hs

```diff
@@ -7,6 +7,12 @@
 #endif
 {-# OPTIONS_GHC -Wall #-}
 #include "free-common.h"
+-- Manual Typeable instances
+#if __GLASGOW_HASKELL__ < 707
```

Comment:
Nitpick: I would prefer to invert the conditional here:

```hs
#if __GLASGOW_HASKELL__ >= 708
{-# LANGUAGE DeriveDataTypeable #-}
{-# LANGUAGE Safe #-}
#else
-- Manual Typeable instances
{-# LANGUAGE Trustworthy #-}
#endif
```

Notice that I'm folding the `DeriveDataTypeable` pragma above into this, as it belongs with the use of `Safe`.

Similar comments apply to other conditional uses of `Trustworthy`/`Safe` elsewhere.

Reply:
Could I remove the `{-# OPTIONS_GHC -Wall #-}`, I think they don't add much value in source files.

Reply:
I didn't even notice that!

I'd be fine with removing `-Wall` there (as I believe it's redundant), although perhaps that's better fit for a separate commit (in the same PR).

Reply:
Bundled it all together. The diff is quite messy (preserving `OPTIONS_GHC -Wall` wouldn't make it any prettier).